### PR TITLE
feat(NavigationBar): new navigation bar components

### PR DESCRIPTION
**Pull request by atabel**
jira: WEB-294
spec: https://app.zeplin.io/project/5d653c69f828bf7299c551c1/dashboard?seid=610d1eead31de315fbce7392

### CODE REVIEW

File: src/navigation-bar.tsx [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+        burgerMenuExit: {
+            transform: 'translate(0)',
+        },
+        burgerMenuExitActive: {
+            transform: 'translate(-100vw)',
+        },
+    };
+});
+
+type MainNavigationBarSection =
```
Comment (yceballost) [inline src/navigation-bar.tsx:261]: Missing hover state in desktop views
https://zpl.io/29dLjWp

File: src/navigation-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+            <>
+                <FocusTrap disabled={disableFocusTrap} group="burger-menu-lock">
+                    <header>
+                        <ThemeVariant isInverse={isInverse}>
+                            <div className={classnames(classes.navbar, {[classes.topFixed]: topFixed})}>
+                                <ResponsiveLayout>
+                                    <Inline space="between" alignItems="center">
+                                        <Inline space={24} alignItems="center">
+                                            <IconButton
+                                                aria-live="polite"
```
Comment (anabelengp) [inline src/navigation-bar.tsx:329]: Is this aria-live working here? is it intended to read the menu options? or the selected option?

File: playroom/snippets.tsx [hunk 3]
```diff
@@ -890,6 +890,80 @@ const exampleScreens: Array<Snippet> = [
...
+        aria-label="shopping cart with 2 items"
+      >
+        <Badge value={2}>
+          <IconShoppingCartRegular color="currentColor" />
+        </Badge>
+      </NavigationBarAction>
+      <NavigationBarAction onPress={() => {}} aria-label="Open profile">
+        <Circle
+          backgroundImage="https://i.pravatar.cc/100?img=31"
+          size={32}
```
Comment (yceballost) [inline playroom/snippets.tsx:917]: I can see in specs, in mobile view the size of the profile photo is 24. Can you confirm this @HectorSancho

File: src/navigation-bar.tsx (hunk 1 above)
Comment (atabel) [reply src/navigation-bar.tsx:272]: done ;)

File: src/navigation-bar.tsx (hunk 2 above)
Comment (atabel) [reply src/navigation-bar.tsx:329]: Yes. It reads the close/open icon button label when it changes

File: src/navigation-bar.tsx [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+                                                aria-label={
+                                                    isMenuOpen
+                                                        ? 'close navigation menu'
```
Comment (atabel) [inline src/navigation-bar.tsx:332]: translation pending

File: src/touchable.tsx [hunk 5]
```diff
@@ -20,6 +20,7 @@ const redirect = (url: string, external = false): void => {
 
 const useStyles = createUseStyles(() => ({
     touchable: {
+        lineHeight: 'inherit',
```
Comment (pladaria) [inline src/touchable.tsx:23]: Is this causing the height difference in rows? Is this expected?

If that's the case, this change will produce *a lot* of screenshot updates in webapp

Comment (pladaria) [reply src/touchable.tsx:23]: what motivated this change?

File: src/navigation-bar.tsx (hunk 4 above)
Comment (atabel) [reply src/navigation-bar.tsx]: done

File: src/touchable.tsx (hunk 5 above)
Comment (atabel) [reply src/touchable.tsx:23]: I need to review this because I may not need this change

Comment (atabel) [reply src/touchable.tsx]: No, I think I don't need it, reverted.

**Review by yceballost — APPROVED** [review]
superb✨

**Review by AnaMontes11 — APPROVED** [review]
It looks good to me!

File: playroom/snippets.tsx (hunk 3 above)
Comment (anabelengp) [reply playroom/snippets.tsx]: Could it be interesting for the future to componentize some of these actions (UserProfile, shoppingCart, help...) in order to release developers from having to know how to configure sizes, colors etc, and make changes easier?

Comment (atabel) [reply playroom/snippets.tsx:917]: changed

File: src/navigation-bar.tsx [hunk 6]
```diff
@@ -0,0 +1,563 @@
...
+            // See: https://hover-pointer-media-query.glitch.me/ and https://github.com/mui-org/material-ui/issues/15736
+            '@media (pointer: fine), (pointer: none)': {
+                '&:hover span': {
+                    color: ({isInverse}) =>
+                        isInverse ? theme.colors.textSecondaryInverse : theme.colors.textSecondary,
+                },
+            },
+        },
+        selectedSection: {
+            borderColor: ({isInverse}) => (isInverse ? theme.colors.inverse : theme.colors.controlActivated),
```
Comment (anabelengp) [inline src/navigation-bar.tsx:236]: should we mark the selected section also in mobile/tablet devices (I think it is not being marked in the menu when it is open)

**Review by anabelengp — APPROVED** [review]

File: playroom/snippets.tsx (hunk 3 above)
Comment (atabel) [reply playroom/snippets.tsx]: >Could it be interesting for the future to componentize some of these actions (UserProfile, shoppingCart, help...) in order to release developers from having to know how to configure sizes, colors etc, and make changes easier?

If we find this is a repetitive pattern, for sure. But I don't think so

**Review by HectorSancho — APPROVED** [review]

File: src/navigation-bar.tsx (hunk 6 above)
Comment (atabel) [reply src/navigation-bar.tsx:236]: Not according design specs. Already discussed with design team

File: src/focus-trap.tsx [hunk 7]
```diff
@@ -1,22 +1,17 @@
 import * as React from 'react';
-import FocusTrapReact from 'focus-trap-react';
+import ReactFocusLock from 'react-focus-lock';
 
 type Props = {
     children: React.ReactNode;
-    clickOutsideDeactivates?: boolean;
```
[the remark this answers is not in the document]
Comment (atabel) [reply src/focus-trap.tsx:6]: I'll integrate this breaking change in a separated branch (https://github.com/Telefonica/mistica-web/pull/327) and (among with other breaking changes) will release a new major

**Pull request merged**